### PR TITLE
Detect OTP >= 21 even if rebar is not used

### PR DESCRIPTION
**Pull request by zuiderkwast**
Fix compilation without Rebar.

Also update mix.exs to allow later versions of Elixir.

### CODE REVIEW

File: include/eredis.hrl [hunk 1]
```diff
@@ -27,11 +27,18 @@
 -type parser_state() :: status_continue | bulk_continue | multibulk_continue | error_continue.
 
 %% Internal types
+-ifdef(OTP_RELEASE).
+%% OTP 21 or later
+-define eredis_queue :: queue:queue().
```
Comment (alexandremcosta) [inline include/eredis.hrl:32]: when trying to compile:
```
==> eredis
warning: the dependency :eredis requires Elixir "~> 1.5.1" but you are running on v1.11.2
Compiling 6 files (.erl)
include/eredis.hrl:32: badly formed 'define'
include/eredis.hrl:32: badly formed 'define'
include/eredis.hrl:32: badly formed 'define'
include/eredis_sub.hrl:19: type eredis_queue() undefined
include/eredis.hrl:32: badly formed 'define'
src/eredis_client.erl:322: type eredis_queue() undefined
include/eredis.hrl:32: badly formed 'define'
```

I tried to change it to
```
-define(eredis_queue, queue:queue()).
```

And it fixed the `badly formed 'define'`, but I still get
```
==> eredis
warning: the dependency :eredis requires Elixir "~> 1.5.1" but you are running on v1.11.2
Compiling 6 files (.erl)
include/eredis_sub.hrl:19: type eredis_queue() undefined
src/eredis_client.erl:322: type eredis_queue() undefined
```

I wasn't able to fix them...

**Review by alexandremcosta — CHANGES_REQUESTED** [review]

File: include/eredis.hrl (hunk 1 above)
Comment (zuiderkwast) [reply include/eredis.hrl:32]: I will fix. Sorry for posting a PR without even trying to build. :-D

Comment (zuiderkwast) [inline include/eredis.hrl:32]:
```suggestion
-type eredis_queue() :: queue:queue().
```

**Comment by zuiderkwast** [comment]
Why this warning:

    warning: the dependency :eredis requires Elixir "~> 1.5.1" but you are running on v1.11.2

Isn't 1.11.2 > 1.5.1?

**Comment by zuiderkwast** [comment]
Is `mix.exs` really needed, considering this is an Erlang lib? Mix can build without it, right?

**Comment by alexandremcosta** [comment]
> Why this warning:
> 
> ```
> warning: the dependency :eredis requires Elixir "~> 1.5.1" but you are running on v1.11.2
> ```
> 
> Isn't 1.11.2 > 1.5.1?

Because `~> 1.5.1` means elixir >= `1.5.0` and < `1.6.0`. To require >= `1.0` and < `2.0` you need `~> 1.5`.

**Comment by alexandremcosta** [comment]
> Is `mix.exs` really needed, considering this is an Erlang lib? Mix can build without it, right?

I believe that if you have the mix file, you make it possible for elixir clients to use this lib.

**Comment by zuiderkwast** [comment]
> To require >= 1.5 and < 2.0 you need ~> 1.5.

Aha! And to require all versions >= 1.5.1, is it possible? Otherwise I guess we should change to ~> 1.5 as you say.

**Comment by alexandremcosta** [comment]
> > To require >= 1.5 and < 2.0 you need ~> 1.5.
> 
> Aha! And to require all versions >= 1.5.1, is it possible? Otherwise I guess we should change to ~> 1.5 as you say.

That's even simpler:
```
      elixir: ">= 1.5.1",
```

**Review by alexandremcosta — APPROVED** [review]
now it works gracefully, thank you so much :)

**Comment by zuiderkwast** [comment]
Great! I'll let @bjosv look at this too. I think we can merge it tomorrow.

Maybe we should ta `1.3.1-nordix` after that. Is there anything else we have missed that should be in 1.3.1?

**Comment by alexandremcosta** [comment]
I just ran StressTest with your fork in our staging environment and it's much more stable then mine under high load and during resharding. We will start using it in production. Please add tags to make dependents life easier until we take over the Hex name.

**Comment by bjosv** [comment]
> Great! I'll let @bjosv look at this too. I think we can merge it tomorrow.
> 
> Maybe we should ta `1.3.1-nordix` after that. Is there anything else we have missed that should be in 1.3.1?

LGTM!
I dont think we currently have anything more, an update to CHANGELOG regarding a 1.3.1 would probably be nice.

**Pull request merged**